accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,211 @@
+
+  - Images - distributed as large tarballs via http
+  - Image metadata - pillars, activation keys, list of repos and packages (in DB)
+  - [HWTYPE](https://www.uyuni-project.org/uyuni-docs/en/uyuni/retail/retail-deploy-terminals.html#_create_a_hardware_type_group) configurations
+
+## Local data
+
+The following items do not need central management and can stay on Peripheral Server (like in the current implementation).
+
+  - Branch groups -  groups containing Branch Server and connected terminals, includes Saltboot groups for Containerized Branch Server
+  - Branch formulas - describe configuration of Branch services - network, pxe, dhcp, dns
+  - Branch PXE data - entries describing how to boot each terminal
+
+# Detailed design
+[design]: #detailed-design
+
+## Images + Image metadata
+
+Image files with corresponding metadata and pillars are already synchronized.
+Further customization is possible via API
+
+## HWTYPE configurations
+
+This is currently implemented as a formula assigned to a group with special name.
+
+FIXME: pick one alternative
+
+Alternative 1:
+Extend retail_yaml to create the groups and formula data via Hub and Peripheral Server API.
+This is relatively small change to the code.
+
+Alternative 2:
+Central creation of groups is covered in [Hub Users RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+This approach is probably more flexible.
+
+Both alternatives will have to cope with authentization and Hub vs. Peripheral server user credentials.
+
+In long term, alternative 2 is clean solution.
+
+Alternative 1 might be used as a short term fix and also to keep compatibility.
+
+
+### YAML configuration
+
+FIXME: optional, related to Alternative 2 in previous section
+
+We have retail_yaml tool for initial configuration of Retail environment from
+[YAML](https://github.com/uyuni-project/retail/blob/master/python-susemanager-retail/example.yml) file.
+[Hub Users RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00061-hub-users.md) introduced similar concept, based on Salt.
+Maybe these parts could be merged.
+
+
+## Combined Peripheral and Branch server
+
+The components from Containerized Branch Server can be re-used also in this scenario.
+
+Required components:
+
+- Saltboot group
+- Cobbler / TFTP   
+- Mapping of branch-local URLs
+- DHCP server
+
+
+### Mapping of branch-local URLs
+
+https://github.com/SUSE/spacewalk/issues/18455
+
+Non-containerized retail branch server uses image pillars to map branch-local image URLs to SUMA server URLs.
+
+Containerized branch server does not have access to pillars, so it currently implements this 
+functionality by apache mod_rewrite. This is quite error-prone, because it expect URL in certain 
+format. It can't work with images on external server.
+https://github.com/SUSE/spacewalk/blob/Manager-4.3/containers/proxy-httpd-image/uyuni-configure.py#L166
```

Comment:
This shouldn't be extracted into self-contained issue? It looks like a general problem and not a hub-specific one.

Reply:
yes, it is self-contained issue, 
https://github.com/SUSE/spacewalk/issues/18455